unit-test/bpnode_app_tests.c

```diff
@@ -0,0 +1,232 @@
+/************************************************************************
+ * NASA Docket No. GSC-18,719-1, and identified as “core Flight System: Bootes”
+ *
+ * Copyright (c) 2020 United States Government as represented by the
+ * Administrator of the National Aeronautics and Space Administration.
+ * All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain
+ * a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ************************************************************************/
+
+/*
+** File: bpnode_app_tests.c
+**
+** Purpose:
+** Coverage Unit Test cases for the BPNode Application
+**
+** Notes:
+** This implements various test cases to exercise all code
+** paths through all functions defined in the BPNode application.
+**
+** It is primarily focused at providing examples of the various
+** stub configurations, hook functions, and wrapper calls that
+** are often needed when coercing certain code paths through
+** complex functions.
+*/
+
+/*
+** Includes
+*/
+#include "bplib.h"
+#include "bpnode_test_utils.h"
+
+/*
+**********************************************************************************
+**          TEST CASE FUNCTIONS
+**********************************************************************************
+*/
+
+/*
+** Test Case For:
+** void BPNode_AppMain( void )
+*/
+void Test_BPNode_AppMain(void)
+{
+    CFE_SB_Buffer_t  Buf;
+    CFE_SB_Buffer_t *BufPtr = &Buf;
+    UT_CheckEvent_t  EventTest;
+
+    /*
+     * BPNode_AppMain does not return a value,
+     * but it has several internal decision points
+     * that need to be exercised here.
+     *
+     * First call it in "nominal" mode where all
+     * dependent calls should be successful by default.
+     */
+    BPNode_AppMain();
+
+    UtAssert_STUB_COUNT(CFE_ES_ExitApp, 1);
+    UtAssert_UINT32_EQ(BPNode_AppData.RunStatus, CFE_ES_RunStatus_APP_RUN);
+
+    /* Failure of BPNode_AppInit()*/
+    UT_SetDeferredRetcode(UT_KEY(CFE_EVS_Register), 1, CFE_EVS_INVALID_PARAMETER);
+
+    BPNode_AppMain();
+
+    UtAssert_UINT32_EQ(BPNode_AppData.RunStatus, CFE_ES_RunStatus_APP_ERROR);
```

Comment:
(This pretty much applies for all of our tests.)
Could you split these into different test cases?

That way, if some of these tests fail and some of them pass, we should be able to see more granular results from the test output.

Reply:
We should still be able to see the granular output, each UtAssert call should print its pass/fail status

Reply:
good enough then, I suppose. I do feel like this method is going to cause us to have some really big test cases though.

Reply:
Reopening due to discussion via Teams

Reply:
Done